lib/auth/init.go

```diff
@@ -264,23 +263,6 @@ func Init(cfg InitConfig, opts ...ServerOption) (*Server, error) {
 		return nil, trace.Wrap(err)
 	}
 
-	// set cluster level config on the backend and then force a sync of the cache.
-	clusterConfig, err := asrv.GetClusterConfig()
-	if err != nil && !trace.IsNotFound(err) {
-		return nil, trace.Wrap(err)
-	}
-	// init a unique cluster ID, it must be set once only during the first
-	// start so if it's already there, reuse it
-	if clusterConfig != nil && clusterConfig.GetClusterID() != "" {
-		cfg.ClusterConfig.SetClusterID(clusterConfig.GetClusterID())
-	} else {
-		cfg.ClusterConfig.SetClusterID(uuid.New())
-	}
-	err = asrv.SetClusterConfig(cfg.ClusterConfig)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
 	// The first Auth Server that starts gets to set the name of the cluster.
 	err = asrv.SetClusterName(cfg.ClusterName)
```

Comment:
update this to re-use the existing `ClusterID` from `ClusterName` stored in the backend

Reply:
That should be covered by `migrateClusterID`. Internally, `SetClusterName` performs a `Create`, not a `Put`, so an existing `ClusterName` is never overwritten -- except for `migrateClusterID` that uses `UpsertClusterName` for that reason.